cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/aggregator/RawMetricValues.java

```diff
@@ -191,15 +191,17 @@ public synchronized void resetWindowIndexes(long startingWindowIndex, int numWin
         || inValidRange(startingWindowIndex + numWindowIndexesToReset - 1)) {
       throw new IllegalStateException("Should never reset a window index that is in the valid range");
     }
-    LOG.debug("Resetting window index [{}, {}]", startingWindowIndex,
-              startingWindowIndex + numWindowIndexesToReset - 1);
     // We are not resetting all the data here. The data will be interpreted to 0 if count is 0.
+    long numAbandonedSample = 0;
     for (long i = startingWindowIndex; i < startingWindowIndex + numWindowIndexesToReset; i++) {
       int index = arrayIndex(i);
+      numAbandonedSample += _counts[index];
```

Comment:
`_counts[index]` is a `short`, whose max value is `(2^15)-1`.`long` has a max value of `(2^63)-1`. If each `_counts[index]` has the max value, this can handle about `(2^48)` entries in `_counts`. Do we need a `long` or would an `int` (i.e. supports max value of `(2^31)-1`) be sufficient -- `64K` entries `_counts` in the worst case?

Reply:
I feel `int` should be enough here. the number of sample we abandon is number of window to reset * `number of sample per window`, I do not expect this to be a huge number.